test/livebook/intellisense_test.exs

```diff
@@ -1053,6 +1053,54 @@ defmodule Livebook.IntellisenseTest do
              ] = Intellisense.get_completion_items("struct.my", binding, env)
     end
 
+    test "completion for struct keys inside struct" do
+      {binding, env} = eval(do: nil)
+
+      assert [
+               %{
+                 label: "my_val",
+                 kind: :field,
+                 detail: "field",
+                 documentation: nil,
+                 insert_text: "my_val"
+               }
+             ] =
+               Intellisense.get_completion_items(
+                 "%Livebook.IntellisenseTest.MyStruct{my",
+                 binding,
+                 env
+               )
+    end
+
+    test "completion for struct keys inside struct removes filled keys" do
+      {binding, env} =
+        eval do
+          struct = %Livebook.IntellisenseTest.MyStruct{}
+        end
+
+      assert [] =
+               Intellisense.get_completion_items(
+                 "%Livebook.IntellisenseTest.MyStruct{my_val: 123, ",
+                 binding,
+                 env
+               )
+    end
+
+    test "completion for struct keys inside struct ignores `__exception" do
+      {binding, env} = eval(do: nil)
+
+      refute [
+               %{
+                 label: "__exception__"
+               }
+             ] =
```

Comment:
I think this assertion is not really strong because if it returns two elements, one of them being `__exception__`, the test will still fail. What do you think about something like this:

    completion = ...
    refute Enum.find(completion, & &1.label == "__exception__")

Reply:
doh!

```
      refute Enum.find(completions, &match?(%{label: "__exception__"}, &1))
```

Reply:
Yes, your suggestion also works!